apps/dashboard/app/models/configuration_property.rb

```diff
@@ -0,0 +1,61 @@
+class ConfigurationProperty
+
+  def self.with_boolean_mapper(name:, default_value: nil, read_from_env: false, env_name: nil)
+    ConfigurationProperty.new(name, default_value, read_from_env, env_name, BooleanMapper)
+  end
+
+  def self.with_integer_mapper(name:, default_value: nil, read_from_env: false, env_name: nil)
+    ConfigurationProperty.new(name, default_value, read_from_env, env_name, IntegerMapper)
+  end
+
+  def self.property(name:, default_value: nil, read_from_env: false, env_name: nil)
+    ConfigurationProperty.new(name, default_value, read_from_env, env_name, PassThroughMapper)
+  end
+
+  attr_reader :name, :default_value, :read_from_environment, :environment_name
+  alias_method :read_from_environment?, :read_from_environment
+
+  def initialize(name, default_value, read_from_env, env_name, mapper)
+    @name = name.to_sym
+    @default_value = default_value
+    @read_from_environment = read_from_env
+    @mapper = mapper
+
+    environment_name = env_name || "OOD_#{@name.to_s.upcase}"
```

Comment:
Using a different `env_name` is going to be confusing should anyone ever decide to use it. Can we get rid of that option?

Or at least, we can add it later if we need. There could be some other scheme for backward comparability.

Reply:
Indeed it looks like this breaks `DISABLE_DASHBOARD_LOGO` moving it to `OOD_DISABLE_DASHBOARD_LOGO`.

Reply:
Sorry, this is me no paying enough attention. It should be `DISABLE_DASHBOARD_LOGO`. Will fix

Reply:
NP. I wonder now how to forward port this. I mean that we'd likely want to prefix this with `OOD_` and respond to it as well. Perhaps even `OOD_` having precedence over the original.

Is that something you can do now? I'm just as happy to defer it.

Reply:
In my first implementation I had support for this use case.
I wanted to support the current use case for `brand_bg_color` that allows for different environment variables.

In the first implementation of `ConfigurationProperty`, the `environment_name` override was an array.

I can put it back in and we can discuss it

Reply:
> use case for brand_bg_color that allows for different environment variables.

Yea I cannot imagine this use case. Which is to say - it confuses me why'd you want several ENV variables.  I'd like to see all configuration move to teh `ondemand.d` directory and really deprecate ENV variables. Mostly for convenience to the admin.  To us (developers) they're nice to have to use, mostly in tests, so we don't have to write files. 

So that's my way of thinking - that actually ENV variables are going away (not really, but for all intents and purposes to the admin/end user of these configs).

Reply:
I am now looking at `brand_link_active_bg_color` which may be what you're thinking. Yea, I'd say that was a bad decision to use several environment variables. There should be only 1 way to do a thing - otherwise documentation and/or administrative overhead is too painful.

And even though in the same breath I'm saying you can have it in `ondemand.d` or through an ENV variable - but again - over time documentation will only really point to `ondemand.d`.

Reply:
For now, shall I add a new method to support the new env name value for `disable_dashboard_logo`?

```ruby
  def disable_dashboard_logo
    disabled = ENV.values_at('OOD_DISABLE_DASHBOARD_LOGO', 'DISABLE_DASHBOARD_LOGO').compact.first
    disabled || fetch(:disable_dashboard_logo)
  end
```

Reply:
Yea that'll be fine. Just to confirm `use case for brand_bg_color that allows for different environment variables.` was using BOOTSTRAP_ prefixed ENV variables?

Reply:
No, the implementation for `ConfigurationProperty` was taking an array for environment name override.
Then, the implementation was grabbing the values in order and taking the first one if any.

```ruby
ConfigurationProperty.property(name: :brand_bg_color, read_from_env: true, env_names: ["OOD_BRAND_BG_COLOR", "BOOTSTRAP_NAVBAR_DEFAULT_BG", BOOTSTRAP_NAVBAR_INVERSE_BG])
```

Reply:
Actually that implementation looks fairly clean - we can go with that if you have it. Indeed - it will be easy to add deprecation warnings because they'll all be in 1 place.

Reply:
Will update